torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -25,58 +26,135 @@ static int ceilDiv(const int a, const int b) {
   return (a + b - 1) / b;
 }
 
+// IO data structure for kernel code;
+static auto typeinfo = R"(
+template<typename T, int N>
+struct Tensor {
+  T& operator[](int ind) {
+    return data[ind];
+  };
+
+  T* data;
+  int size[N];
+  int stride[N];
+};
+)";
+
+// include IO data structure for host code
+struct KernelArgumentHolder {
+  std::vector<void*> arguments;
+  std::vector<char> buffer;
+  void* buffer_ptr;
+
+  KernelArgumentHolder(size_t n_inputs, size_t n_dimension_per_tensor) {
+    arguments.reserve(n_inputs);
+    // We are being generous here on the allocated buffer;
+    buffer.resize(
+        n_inputs * (sizeof(void*) + 2 * sizeof(int) * n_dimension_per_tensor));
```

Comment:
`sizeof(size_t)`

Reply:
Will change along with the struct above.